main.go

```diff
@@ -154,3 +220,197 @@ func gatekeeperNamespace(platformInfo platform.PlatformInfo) (string, error) {
+		time.Sleep(time.Second)
+		return c.isApiAvailable(mainCtx, result, dynamicClient)
+	}
+
+	for _, condition := range conditions {
+		conditionObj, ok := condition.(map[string]interface{})
+		if !ok {
+			continue
+		}
+
+		conditionStatus, ok := conditionObj["status"].(string)
+		if !ok {
+			continue
+		}
+
+		conditionType, ok := conditionObj["type"].(string)
+		if !ok {
+			continue
+		}
+
+		if conditionType == string(v1.NamesAccepted) && conditionStatus == "True" {
+			return true
+		}
+	}
+
+	setupLog.V(1).Info("Constraintpodstatuses crd is not ready, Wait for crd status update")
+
+	time.Sleep(time.Second)
+	return c.isApiAvailable(mainCtx, result, dynamicClient)
+}
+
+// Watch the constraintpodstatuses resource in this function. When adding or Modified events comes,
+// This function adds constraintpodstatus function. Modified events are expected as status changes
+// When deleting event comes, close the function using context
+func (c *ConstraintStatusInstaller) watchGatekeeperInstall(mainCtx context.Context, dynamicClient *dynamic.DynamicClient,
+	cfg *rest.Config, managerOptions manager.Options, namespace string,
+) {
+	var ctx context.Context
+	var ctxCancel context.CancelFunc
+
+	watcher, _ := c.newWatcher(mainCtx, dynamicClient)
+	for {
+		select {
+		case result, ok := <-watcher.ResultChan():
+			// the channel got closed, so we need to restart
+			if !ok {
+				// main context closed. This loop should be closed.
+				if mainCtx.Err() != nil {
+					ctxCancel()
+					return
+				}
+				watcher, _ = c.newWatcher(mainCtx, dynamicClient)
+				continue
+			}
+
+			switch result.Type { //nolint:exhaustive
+			case watch.Modified, watch.Added:
+				if c.getIsInstalled() {
+					continue
+				}
+
+				// Check constraintpodstatuses crd available
+				if !c.isApiAvailable(mainCtx, result, dynamicClient) {
+					break
+				}
+
+				ctx, ctxCancel = context.WithCancel(mainCtx)
+				c.setIsInstalled(true)
+
+				go func() {
+					mgr, err := ctrl.NewManager(cfg, managerOptions)
+					if err != nil {
+						setupLog.Error(err, "unable to setup ConstraintStatus manager")
+					}
+					// Start with new context since previous context was cancelled
+					err = c.addConstraintStatusManager(ctx, mgr, dynamicClient, namespace)
+					if err != nil {
+						setupLog.Error(err, "unable to start ConstraintStatus manager")
+					}
+				}()
+
+			case watch.Deleted:
+				c.setIsInstalled(false)
+				ctxCancel()
+				setupLog.Info("Gatekeeper deleted")
+			}
+		case <-mainCtx.Done():
+			// When parents ctx close, then child should be closed
+			ctxCancel()
+			return
+		}
+	}
+}
+
+func (c *ConstraintStatusInstaller) newWatcher(mainCtx context.Context,
+	dynamicClient *dynamic.DynamicClient,
+) (*toolsWatch.RetryWatcher, error) {
+	timeout := int64(30)
+	gatekeeperGVR := schema.GroupVersionResource{
+		Group:    "operator.gatekeeper.sh",
+		Version:  "v1alpha1",
+		Resource: "gatekeepers",
+	}
+
```

Comment:
Watcher changed. now it watch the gatekeeper resource

Reply:
So in the gap of time between when the controller starts and when Gatekeeper is installed, are there just warning logs about the watch from controller-runtime on constraintpodstatus not starting?

Reply:
I added
```
	if err != nil {
		setupLog.V(1).Info("audit is not set. ConstraintStatus controller cannot be created")
		return false
	}

	auditFromCache := audit["auditFromCache"]
	if auditFromCache != string(operatorv1alpha1.AuditFromCacheAutomatic) {
		setupLog.V(1).Info("auditFromCache is not Automatic. ConstraintStatus controller cannot be created")
		return false
	}
```

Reply:
Also I added this too
> setupLog.Info("ConstraintPodStatus controller is not started yet")